connman/src/firewall-iptables.c

```diff
@@ -2697,98 +2657,72 @@ static int enable_new_firewall_rules(struct connman_service *service,
 
 static int firewall_reload_configurations()
 {
-	GError *error = NULL;
-	GDir *dir;
-	GSList *read_files = NULL;
-	GSList *slist_iter = NULL;
+	GList *read_files = NULL;
 	GList *list_iter = NULL;
 	GHashTableIter iter;
 	gpointer key, value;
 	struct connman_service *service;
 	enum connman_service_type type;
 	struct firewall_context *ctx;
-	const char *filename;
 	const char *config_file;
 	char *ifname;
-	char *filepath;
 	bool new_configuration_files = false;
 	int err = 0;
 
 	/* Nothing to read */
 	if (!g_file_test(FIREWALLCONFIGDIR, G_FILE_TEST_IS_DIR))
 		return 0;
 
-	dir = g_dir_open(FIREWALLCONFIGDIR, 0, &error);
-
-	if (!dir) {
-		if (error) {
-			DBG("cannot open dir, error: %s", error->message);
-			g_clear_error(&error);
-		}
-
-		/* Ignore dir open error in reload */
-		return 0;
-	}
-
-	DBG("read configs from %s", FIREWALLCONFIGDIR);
-
-	/* Read filenames into ordered list */
-	while ((filename = g_dir_read_name(dir))) {
-		/* Read configs that have firewall.conf suffix */
-		if (!g_str_has_suffix(filename, FIREWALLFILE))
-			continue;
-
-		/*
-		 * Add file name to read file list for checking if config file
-		 * has been removed. At this point ignore file tests.
-		 */
-		read_files = g_slist_prepend(read_files, g_strdup(filename));
+	/* Read the filenames only and do not use callback to process */
+	err = util_read_config_files_from(FIREWALLCONFIGDIR, FIREWALLFILE,
+					&read_files, NULL);
+	switch (err) {
+	case -EINVAL:
+	case -ENOMEM:
+		connman_warn("Failed to read configs from %s: %s",
+						FIREWALLCONFIGDIR,
+						strerror(-err));
+		return err;
+	case -ENOTDIR:
+	case -EACCES:
+		connman_warn("%s: %s, continue reloading firewall",
+						FIREWALLCONFIGDIR,
+						strerror(-err));
```

Comment:
after seeing N of these: Is there some profound reason why it would be preferable to have this kind of code duplication everywhere instead of doing such logging from within util_read_config_files_from()?

Reply:
Well yeah the warnings can be moved, however this is the reload case that is different to normal loading here. Maybe that upper warning needs a better description. Not sure what to change, the different cases here kind of justify this..